config/config.go

```diff
@@ -1301,3 +1307,28 @@ func (c *RemoteWriteConfig) UnmarshalYAML(unmarshal func(interface{}) error) err
 	}
 	return nil
 }
+
+// RemoteReadConfig is the configuration for reading from remote storage.
+type RemoteReadConfig struct {
+	URL           *URL           `yaml:"url,omitempty"`
+	RemoteTimeout model.Duration `yaml:"remote_timeout,omitempty"`
+	BasicAuth     *BasicAuth     `yaml:"basic_auth,omitempty"`
+	TLSConfig     TLSConfig      `yaml:"tls_config,omitempty"`
+	ProxyURL      URL            `yaml:"proxy_url,omitempty"`
```

Comment:
Not ad-hoc; some effort went into unifying the http client code (at least, the TLS handling) between the remote write path and the retrieval path: https://github.com/prometheus/prometheus/pull/1957

The HTTPClientConfig struct was extract in #2215, AFAICT to unify the alert manager config and the retrieval config.  I think it would make sense to use it for the remote read/write config too - am happy to put together a PR for that @juliusv?